### PR TITLE
Fix cpu peak of auth secret watcher

### PR DESCRIPTION
**Pull request by guangbochen**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

harvester process spike to ~50% CPU after a period of time.

**Solution:**

The k8s server will timeout the watch connection after 30-60 minutes, while the watch loop in `auth_secret_watcher` does not handle the closed channel correctly and it will lead to an infinity call to raise a CPU peak.

pprof_view

**Related Issue:**
https://github.com/rancher/harvester/issues/424

**Test plan:**

1. using a custom builder image and validating the harvester CPU will stay steady after a period of time.
2. make sure when the users apply a new auth secret of `harvester-key-holder` the `auth_secert_watcher` is called even after the default watch timeout(default to 30 minutes).

### CODE REVIEW

File: pkg/controller/global/auth/auth_secret_watcher.go [hunk 1]
```diff
@@ -28,25 +28,50 @@ func WatchSecret(ctx context.Context, scaled *config.Scaled, namespace, name str
...
-	}
 
+OUTER:
 	for {
-		select {
-		case watchEvent := <-watcher.ResultChan():
-			if watch.Modified == watchEvent.Type {
-				if sec, ok := watchEvent.Object.(*corev1.Secret); ok {
+		logrus.Debugf("Starting auth secret watcher of %s:%s", namespace, name)
+		watcher, err := secrets.Watch(namespace, opts)
```
Comment (gitlawr) [inline pkg/controller/global/auth/auth_secret_watcher.go]: Can we refactor to use informer?

**Comment by guangbochen** [comment]
@gitlawr updated and verified with custom build image.
image

**Comment by gitlawr** [comment]
LGTM. Can you do a rebase?

**Comment by guangbochen** [comment]
@gitlawr  updated, thanks

**Review by gitlawr — APPROVED** [review]

**Pull request merged**